Add SciPy 2020 pyhf proposal (#2)
matthewfeickert/SciPy2020-Proposal: README.md

# [SciPy 2020](https://www.scipy2020.scipy.org/) Submission Instructions

## Author 1

- First Name: Matthew
- Last Name: Feickert
- Email: matthew.feickert at-no-spam cern.ch
- Country: United States
- Organization: University of Illinois at Urbana-Champaign
- Web page: http://www.matthewfeickert.com/
- Corresponding Author: Yes

## Author 2

- First Name: Lukas
- Last Name: Heinrich
- Email: lukas.heinrich at-no-spam cern.ch
- Country: Germany
- Organization: CERN
- Web page: http://www.lukasheinrich.com/
- Corresponding Author: No

## Author 3

- First Name: Giordon
- Last Name: Stark
- Email: gstark at-no-spam cern.ch
- Country: United States

> - Organization: University of California Santa Cruz Santa Cruz Institute for Particle Physics (UCSC SCIPP)

Reviewer: ```suggestion
- Organization: Santa Cruz Institute for Particle Physics, UC Santa Cruz
```

> - Web page: https://giordonstark.com/

Reviewer: ```suggestion
- Web page: https://giordonstark.com/?utm_source=scipy&utm_medium=proposal&utm_campaign=2020
```

Author: Nice.

- Corresponding Author: No

## Title and Abstract
> The title and the abstract should be entered as plain text, they should not contain HTML elements.

### Title

> pyhf: a pure Python statistical fitting library with tensors and autograd

Reviewer: ```suggestion
pyhf: a Python-only statistical fitting library with tensors and autograd
```

Author: Why the change here? We've used "pure Python" to describe it in all work in the past.

### Abstract

> c.f. the [Planning for your proposal submission?](https://github.com/matthewfeickert/SciPy2020-Proposal/blob/master/info_page.md#planning-for-your-proposal-submission) information page

> Your placement in the program will be based on reviews of your abstract.
This should be a roughly 500 word outline of your presentation.
This outline should concisely describe software of interest to the SciPy community, tools or techniques for more effective computing, or how scientific Python was applied to solve a research problem.
A traditional background/motivation, methods, results, and conclusion structure is encouraged but not required.
Links to project websites, source code repositories, figures, full papers, and evidence of public speaking ability are encouraged.

#### Introduction and Motivation

In experimental high energy physics (HEP), statistical fitting tools have been implemented almost entirely in C++.
One common example used by experimentalists at CERN uses the "HistFactory" p.d.f. template.

> Prominently used in the 2012 discovery of the Higgs boson by the ATLAS and CMS collaborations, the p.d.f. template has only existed within the C++ based ROOT framework developed and maintained by CERN.

Reviewer: ```suggestion
Prominently used in the 2012 discovery of the Higgs boson by the ATLAS and CMS collaborations, the p.d.f. template has only been implemented within the C++ based ROOT framework, maintained by CERN.
```

pyhf is a pure-Python implementation of that statistical model for multi-bin histogram-based analysis with asymptotic interval estimation, and part of the Scikit-HEP project (https://scikit-hep.org/) ecosystem.
pyhf supports modern computational graph libraries as computational backends in order to make use of features such as auto-differentiation and GPU acceleration.

> Additionally, the statistical models are defined in an expressive JSON schema, readily enabling preservation and distribution through services such as the Durham High-Energy Physics Database (HEPData) (https://www.hepdata.net/).

Reviewer: ```suggestion
Additionally, the statistical models are defined in a declarative JSON schema, readily enabling preservation and distribution through services such as the Durham High-Energy Physics Database (HEPData) (https://www.hepdata.net/).
```

#### Performance gain through tensorization

> 

Author: We're already over the suggested 500 words, so I've just added a small comment.

> Through adoption of open source "tensor" computational Python libraries, pyhf decreases the abstractions between a physicist analyst and the statistical modeling without sacrificing computational speed.

Reviewer: ```suggestion
Through adoption of open source "tensor" computational Python libraries, pyhf decreases the abstractions between a physics analysis and the statistical modeling without sacrificing computational speed.
```

Author: I really do mean an analyst here &mdash; a person &mdash; not an analysis.

> In fact, through transforming computational graph complexity to tensor dimensionality, pyhf outperforms the traditional C++ implementations on data from real LHC analyses.

Reviewer: ```suggestion
In fact, by taking advantage of tensor calculations, pyhf outperforms the traditional C++ implementations on data from real LHC analyses.
```

pyhf's default computational backend is built from NumPy and SciPy, and supports TensorFlow, PyTorch, and JAX as alternative backend choices.
These alternative backends support hardware acceleration on GPUs, and in the case of JAX JIT compilation, as well as auto-differentiation allowing for calculating the full gradient of the likelihood function --- all significantly speeding up fits.

#### Model specification

pyhf's model specification is in JSON as it offers long term support, is human and machine readable, and is highly compressible.

> All of these benefits make it ideal for analysis archival, and as almost every language has a JSON parser allows for the model specification to be independent of the HistFactory implementation language.

Reviewer: ```suggestion
All of these benefits make it ideal for analysis archival and, since almost every language has a JSON parser, allows for the model specification to be independent of the HistFactory implementation language.
```

#### Conclusions

> pyhf is the first pure-Python implementation of the HistFactory statistical model from HEP.

Reviewer: ```suggestion
pyhf is the first Python-only implementation of the HistFactory statistical model for HEP.
```

> It is being used in HEP by both experimentalists and theorists to perform analyses of LHC data and is enabling open publication of LHC experiment data products for the first time.

Reviewer: ```suggestion
It is being used by experimentalists and theorists analyzing LHC data and is enabling open publication of LHC experiment data products for the first time.
```

Reviewer: "open publication" isn't clear

#### Audience

- Who is the intended audience for your talk?

> This talk would be of particular interest to scientists (experimentalists and theorists alike), engineers, and statisticians (both Frequentist and Bayesian inference can be performed with pyhf) who are looking to use machine learning frameworks for statistical analysis in Python.

Reviewer: ```suggestion
This talk would be of particular interest to scientists, engineers, and statisticians (both Frequentist and Bayesian inference can be performed with pyhf) who are looking to use machine learning frameworks for statistical analysis in Python.
```

- What, specifically, will attendees learn from your talk?

> Regardless of audience background, they will also see how they can use JSON to define an expressive human and machine readable schema for statistical models and implement those models in modern machine learning frameworks.

Reviewer: not sure this is the best takeaway

Author: Revised this a bit.
